src/odemis/driver/piezomotor.py

```diff
@@ -1159,5 +1159,5 @@ def _do_move(self, steps):
                 return
             else:
                 time.sleep(0.1)
-        self.current_pos = copy.deepcopy(self.target_pos)
+        self.current_pos = target_pos
```

Comment:
I can imagine that passing `target_pos` helps because maybe `self.target_pos` has changed while the move is happening. However, I don't think we can drop the `deepcopy()` . Now both variables are going to point to the same dict. So every call to "E" will directly update the current_pos (as well as target_pos). I doubt this is right.

Reply:
I will put back the `deepcopy`